torch/csrc/distributed/c10d/FileStore.cpp

```diff
@@ -255,7 +256,11 @@ off_t refresh(
     while (size > pos) {
       file.read(tmpKey);
       file.read(tmpValue);
-      cache[tmpKey] = std::move(tmpValue);
+      if (tmpKey.compare(0, deletePrefix.size(), deletePrefix) == 0) {
+        cache.erase(tmpKey.substr(deletePrefix.size()));
```

Comment:
I assume this means if the user unintentionally happens to have a key prefixed "-" this logic breaks and the key gets deleted unexpectedly?

Reply:
No, I don't think this is a problem. Because all user-given keys are prefixed with `/` so it will become `/-somekey`. But thanks for pointing out this, I modified my test to test on this case to double-check it is actually working.